IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -1035,17 +1077,41 @@ - (BOOL)saveAccount:(MSIDAccount *)account
     
     for (MSIDAccountCacheItem *accountCacheItem in allAccounts)
     {
-        // If we have accountIds to filter by, only return account if it has an associated refresh token
-        if ([accountIds containsObject:accountCacheItem.homeAccountId])
+        BOOL shouldReturnAccount = NO;
+        if (signedInAccountsOnly)
         {
-            if (authority.environment)
+            if ([accountIdsFromRT containsObject:accountCacheItem.homeAccountId])
             {
-                accountCacheItem.environment = authority.environment;
+                shouldReturnAccount = YES;
             }
             
-            MSIDAccount *account = [[MSIDAccount alloc] initWithAccountCacheItem:accountCacheItem];
-            if (!account) continue;
-            
+            MSIDAccountMetadataState signInState = [accountMetadataCache signInStateForHomeAccountId:accountCacheItem.homeAccountId clientId:clientId context:nil error:nil];
+            if (signInState == MSIDAccountMetadataStateSignedIn)
+            {
+                shouldReturnAccount = YES;
+            }
+            else if (signInState == MSIDAccountMetadataStateSignedOut)
+            {
+                shouldReturnAccount = NO;
+            }
+        }
+        else
+        {
+            shouldReturnAccount = YES;
+        }
+        
+        // init account from account cache item
+        if (authority.environment)
+        {
+            accountCacheItem.environment = authority.environment;
+        }
+        
+        MSIDAccount *account = [[MSIDAccount alloc] initWithAccountCacheItem:accountCacheItem];
+        if (!account) continue;
+        
+        // add account to the correct set
+        if (shouldReturnAccount)
```

Comment:
if shouldReturnAccount == NO, no need to create MSIDAccount

Reply:
In my implementation, I rely on the "isEqual" method of `MSIDAccount` to do the account "collapse" even for signed-out accounts. That's why i want to create a `noReturnAccountsSet`. What do you think?

Reply:
That makes sense :)